src/export/sealed.rs

```diff
@@ -0,0 +1,82 @@
+#[allow(unused_imports)]
+use crate as defmt;
+use crate::{Format, Formatter, Str};
+
+pub trait Truncate<U> {
+    fn truncate(self) -> U;
+}
+
+macro_rules! impl_truncate {
+    ($t:ty, $u:ty) => {
+        impl Truncate<$t> for $u {
+            fn truncate(self) -> $t {
+                self as $t
+            }
+        }
+    };
+}
+
+impl_truncate!(u8, u8);
+impl_truncate!(u8, u16);
+impl_truncate!(u8, u32);
+impl_truncate!(u8, u64);
+impl_truncate!(u8, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u16, u16);
+impl_truncate!(u16, u32);
+impl_truncate!(u16, u64);
+impl_truncate!(u16, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u32, u32);
+impl_truncate!(u32, u64);
+impl_truncate!(u32, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u64, u64);
+impl_truncate!(u64, u128);
+
+// needed so we can call truncate() without having to check whether truncation is necessary first
+impl_truncate!(u128, u128);
```

Comment:
The comment can probably be deduplicated and rewritten to be a bit clearer with the new macros, maybe something like

> We implement `Truncate<X> for X` so that the macro can unconditionally use it, even if no truncation is performed

Reply:
In addition, this macro could also take a list, and an operator indicating the direction of the truncation would make it a bit clearer what's going on:

```
impl_truncate!(u8 -> u8, u16 -> u8, ...);
```